### DISCUSSION TITLE
Supporting Scene Structure

### DISCUSSION THREAD

**Discussion by instancezero** [vkbo/novelWriter#1769] [Ideas]
Please take a look at the storygrid branch of https://github.com/instancezero/novelWriter

Background: I have completed the first draft of my novel using Scrivener. My first step
in getting to the second draft involves an analysis of the current scenes for good story
structure. Unfortunately, the only way to achieve this is by creating separate text
elements or documenting each scene in a spreadsheet. I'd much rather have the relevant
metadata in the same file as the prose. Since NovelWriter already has a good deal of
what's required for this, I extended it to add this capability.

I don't have deep Python experience, but I've got a lot of coding experience including
a little Qt, so if anything looks weird, put it down to that.

This branch adds several features intended to support developmental editing, using
elements taken from a number of methods including "The Story Grid" by Shawn Coyne.

Several new scene structure classifiers have been added to comments:

- "climax": The climax of the scene
- "complication": The complicating factor(s)
- "crisis": The scene's crisis
- "duration": Elapsed time of the scene
- "incite": the inciting incident
- "polarity": The polarity shift in the scene
- "resolution": The resolution (or non-resolution) of the crisis
- "shift": The value shift in the scene
- "turning": The turning point
- "when": The date/time/ time of day for the scene

A structure button has been added to the document viewer. When enabled, comments
using the new classifiers will appear as a group.

The build dialog has been extended to add two targets for scene structure: as a CSV
file and as a JSON file. The generated file contains one line/entry for each scene
in the file, along with columns containing the scene name, any of the new classifiers
that have been used, along with relevant elements that are already part of 
NovelWriter: plot, synopsis, pov, focus, characters, objects, custom data, and
comments. Any unused elements are excluded from the export.

The exports provide an excellent way to evaluate the overall story structure. The
formats are intened to streamline imports into spreadsheets and databases.

If you have any interest in merging this into the main branch, let me know. I'll be
happy to update the documentation if you like the new features.

**Comment by vkbo** [comment]
Interesting. I tend to try to avoid adding features this specific. I have a feature ticket to implement a general comment style called "Note" which can take a note type as an argument.

The diff is also out of sync with my dev branch, so I can't quite see what is yours and what is my code. I'm not sure where in the GUI this appears in the end and exactly what it does.

I do also track plot points like turning points and crisis, etc. However, I just create `@plot` tags for the purpose under specific headings in my main plot file. They too show up in the Outline and in exported CSV.

**Reply by vkbo** [reply]
Ah, I see now. They are additional text columns in the outline. That is interesting. A bit sparse perhaps though. I wonder if instead of one new syntax per topic this can indeed work with what I've planned for the "Note" implementation.

**Comment by instancezero** [comment]
I've tried to incorporate both the specifics of Story Grid and those of several other less formalized methodologies, so an author can go in as deep as they wish with whatever system they choose (I won't be following Story Grid strictly myself). As for the GUI, here's a couple of captures to illustrate:
![image](https://github.com/vkbo/novelWriter/assets/2599327/c2fa5e51-728d-453a-bc53-15cef1b732d3)
![image](https://github.com/vkbo/novelWriter/assets/2599327/ded5e6da-b9a1-4515-afd6-e2d322ec10ee)

**Reply by vkbo** [reply]
Thanks. I think this is an interesting idea. I do like to support various workflows.

I think this particular implementation is a little too maximalist for novelWriter, which tries to be on the minimalist side of the spectrum. However, it might not take a lot of effort to implement it in a different way.

I think perhaps the notes feature is not the way to go, but maybe something similar. I definitely think you're on the right track using a comment syntax here though.

The notes feature will implement a way to define a subcategory of notes that can be filtered on manuscript build. Perhaps compacting your list of options to just `%Story` with a modifier in a similar fashion would be enough. The processing could be the same as what you have, and auto-completion could help with the modifier words. The framework I added for auto-completing tags can be used here as well. that way you can define all the story grid categories you want to, and not have to extend the list to support all the different methods out there. I prefer to not lock novelWriter down too much into specific writing techniques.

The export of the CSV may be more suitable to add in the Outline view rather than the Manuscript tool, although I'm uncertain exactly what is exported into this CSV.

**Reply by instancezero** [reply]
I may need more coffee, sorry.

Yes, %Story sounds like a good idea (or although it's longer, %Structure, which is the term I used in the code to describe the aggregate). As a side benefit, it would make it easier to assign a Ctrl-K keyboard shortcut using a single character.

I would call the approach embracing rather than maximalist. I did try to incorporate several of the methodologies I've seen because I don't want to be prescriptive, as in "Here's the Story Grid method, use that".

Different editors will have different approaches so I wanted to be able to have this facility adapt to the individual. For this reason, only the terms that have been used will appear in the export. If someone doesn't use %shift, then it won't be generated. I've included a sample CSV and JSON that illustrate this. Note that the character(s) column separates multiple characters with a linefeed so you'll have to set optimal line height to see them all.

Here's an on-the-fly idea: how about letting people define the terms they use in a "Structure" folder, similar to the existing Characters and locations? That would make it so the terms aren't hard-coded, simplifying it in the process.

In my idealized world, authors and editors both manage to learn enough git that a manuscript can be stored in a repo somewhere and can collaborate without passing multiple versions of ODT/word files back and forth.

[Sample Project - My Manuscript tweaked.csv](https://github.com/vkbo/novelWriter/files/14724823/Sample.Project.-.My.Manuscript.tweaked.csv)
[Sample Project - My Manuscript tweaked.json](https://github.com/vkbo/novelWriter/files/14724825/Sample.Project.-.My.Manuscript.tweaked.json)

**Reply by vkbo** [reply]
> I may need more coffee, sorry.

No worries, the user interface here is horrible. If I click "quote" it even opens in the wrong box. GitHub doesn't seem to care to fix the bugs either.

Anyway, what I meant by maximalist is the approach to include new syntax for each single possible option. This rapidly becomes unmaintainable, hence I'm looking for a way to only add the core feature, and make the rest up to the user.

Every time a document is saved, it is passed through the NWIndex class. It shouldn't be difficult to keep a record of all the terms used in Story/Structure and make them available in the editor in an auto-complete menu. We just need a syntax that is as easy as possible. I would also like this feature to be general enough that it could track anything you want.

To be honest, we don't even have to formalise it into a Story/Structure prefix. The code could easily recognise Synopsis and Short like it does, and collect all other `%Whatever:` entries as whatever the "whatever" word is and collect them in a new index object. Anything that doesn't have a term directly after the `%` can be processed as a plain comment. That way, you can define anything you want and add it to the various outputs as meta data by simply starting to write. That is definitely flexible. Auto-completion can help make it consistent.

**Reply by vkbo** [reply]
Some users have also asked for a free for all approach to the `@char` etc type keywords, but that's much more difficult because they are processed in much more complicated ways. This is not the case for comments as they are not processed at all, just collected and displayed in a few select places. Since all the logic is essentially pass-through, I don't see the same issues with just opening it up as a free syntax.

**Reply by vkbo** [reply]
Another point to add is that the syntax highlighter can distinguish between pre-defined comment types and custom terms. This will help further with the usage. I would like to at least reserve "Synopsis", "Short", "Summary" (all as aliases to the same index entry) and "Note" which i would like to format in a different colour in output, and possibly export as actual notes in ODT docs.

The only issue I see is if we add further pre-defined terms and users who've already used them have the functionality change on them. This could possibly be avoided by slightly changing the comment syntax, but it would have to be something that is very easy to use. Like a single character prefix. Like `%+Term:`, `%%Term:` or something along those lines. I'm not sure here. Just throwing out ideas. I don't want the syntax to become too esoteric either.

**Reply by instancezero** [reply]
Indeed. I came close to suggesting an `@story` construct, but the complications were evident. I'm keenly aware of the fact that writers aren't coders, so the syntax has to be as easy to understand as possible. Maybe something like `%story.term:` would do it? Actually, I kind of like `%%Term` just because it would be easy to type.

**Reply by vkbo** [reply]
Yeah, as a professional developer I am a bit wary of the common problem that what makes sense to a developer (especially when used to IDEs) does not necessarily make sense to a regular user. novelWriter is already inspired by IDEs, and uses a markup syntax that is tricky enough for people unfamiliar with Markdown or restructuredText, etc. Having menu entries for everything lowers the threshold a bit I hope.

At the end of the day you have to like the ease of, and limitations of, Markdown as opposed to rich text editors with a thousand formatting buttons that never look exactly like you want, to want to use novelWriter in the first place. But I also try to not get carried away. I was very uncertain if I should add the shortcodes because of this, but there was no other way to get the additional features people kept asking for without introducing more advanced syntax. Most people won't need it though, which is why I ended up adding it. Adding more complex syntax is probably fine, as long as it isn't needed often and you can get most things done without it. My own writing projects contain were little beyond headings, basic tags, and text. Most of the rest of the syntax is rarely used.

Anyway, I'm kind of liking the `%Story.Term:` syntax you propose here. It keeps control of the first word with novelWriter, and makes the second part free for the user. The dot can trigger the auto-complete, just like in an IDE. It also means that if the user just writes `%Story:`, they can use the feature as a single entry if they don't need the added complexity. It also works well for what I intended for `%Note:`. They can also use plain comments if they don't even need the above complexity for tracking meta data. Essentially, you can do what you want in multiple layers of detail. I like that principle.

**Reply by instancezero** [reply]
Draw the line at adding TeX ;) Given that the target audience is **novel** writers, I think much complexity in terms of formatting is not going to be missed. I have one story that makes occasional use of a fixed-pitch font to indicate something written on a display. As far as typography is concerned, I think that might be the only thing missing. I would expect that even then, that can be achieved after exporting to ODT or with one of the several other tools available for formatting a manuscript.

Elsewhere I have ranted at some length about the merits of scalable user interfaces: simple for those with basic needs, but yet powerful enough for those with more complex requirements.  `%Story[.Term]:` is consistent with that. I'll look at refactoring what I have to use that syntax, although what I should be doing right now is a developmental edit. ;)

**Reply by vkbo** [reply]
> I have one story that makes occasional use of a fixed-pitch font to indicate something written on a display. As far as typography is concerned, I think that might be the only thing missing.

Feel free to add a feature request for that. I've been thinking of adding it as a shortcode. Having a task to track it will help remind me.

> Elsewhere I have ranted at some length about the merits of scalable user interfaces: simple for those with basic needs, but yet powerful enough for those with more complex requirements.

Yes, I like that approach too, so it's pretty much what I'm aiming for here.

> `%Story[.Term]:` is consistent with that. I'll look at refactoring what I have to use that syntax, although what I should be doing right now is a developmental edit. ;)

I think a feature request on this too would be useful. I'll write something up based on this discussion, and we can add specifics there,

**Comment by peter88213** [comment]
An interesting proposal and an interesting discussion. Considering how many different methods there are for "story grids", the suggestion of a separate `story` category with user-defined terms seems very sensible to me. 

However, as far as the start time and duration of a scene mentioned at the beginning are concerned, I suggest a more binding definition, both for the category and for the data format. This would open the door for third-party scripts that synchronize the project with dedicated timeline software, such as [Timeline](https://thetimelineproj.sourceforge.net/about.html). 

As an example, I'll mention the [yw-timeline](https://peter88213.github.io/yw-timeline/) Python script that I wrote a while back for yWriter. It's not out of the question that someone might want to program something like this for novelWriter, is it?

**Reply by vkbo** [reply]
Yes, the timeline issue has been on my mind ever since the beginning when I planned novelWriter during lunch breaks with my then colleague @Number042. It is still there in the project's oldest open feature ticket, #76.

The reason there hasn't been much traction on it is that it's difficult to arrive at a sensible solution. The Outline View is the closest attempt yet, but it doesn't really fulfil the request. #854 also proposes an approach to this topic.

I do want to revisit this topic soon as I'm almost done with implementing a ton of easier features, and would really like to get some traction here. The issue for me is that I cannot plan this entirely on my own, because it's a complex task. I would very much like to collaborate with some people on its design. I can write the code, no problem.

One issue that does bother me is that the `@time` keyword is already taken and intended used in a way that is in reality redundant because it is already covered by `@plot`. However, I think it can be repurposed by preserving the current tag/reference meaning and add a way to indicate a timeline/time range to it.

The display of the Timeline tool is similar to what I have in mind, and perhaps a way to integrate with it would be great too. Still, I do want at least some way to visualise it in novelWriter itself.

**Reply by peter88213** [reply]
The Timeline developers have provided a way to embed the Timeline canvas in other applications, but in the case of novelWriter this will probably fail due to the different GUI framework (as far as I know, they use wxPython). 
Then there is apparently the option of programming [File Exporter as a Timeline plug-in](https://thetimelineproj.sourceforge.net/howto-export.html). Although this is only a one-way option, it may be useful for plotting a scene structure.

Developing the whole thing yourself is an appealing prospect, but as you can see from the Timeline project, it's a whole lot of work and full of pitfalls. A project in itself, I would say.

There is also commercial timeline software such as Aeon that is far superior to Timeline, but does not have a data format that is suitable for synchronization.

**Reply by instancezero** [reply]
I would propose that with the `%Story.Term` notation, it would be possible to do some syntax checking. My intention with "duration" was to use a `<number><units>` syntax (eg. 15m for 15 minutes, 3d for 3 days. Dang-it-all English for making months start with m). In this case  `%Story.Start` could require a date that can be parsed successfully. Then NovelWriter could export that data, either in a format Timeline can use or a separate tool could transform something like a JSON export into a file Timeline understands.

**Reply by peter88213** [reply]
The special feature of novelWriter is that all metadata is always explicitly included in the text. Other writing software may operate with a timestamp in the case of date/time information, from which the user interface then generates a localized display format. 
I can imagine that not everyone will be happy with specifically "English" date/time/duration information. The ISO format might be used as a compromise, if semantics are required. 

On the other hand--and here I contradict my suggestion above--the date/time information in free form as a plain string also has its advantages. So you can use it very well e.g. for fantasy calendar items.

**Reply by instancezero** [reply]
I don't know enough Python to answer this but one hopes there are libraries designed to tackle the "all dates look different, even if you only use English" problem. PHP, for example, has packages that will read just about anything in any of several languages and turn that into a normalized date, the bloody d/m/y vs. m/d/y ambiguities aside.

If such libraries exist, then authors could continue to use whichever format they prefer, possibly with a configuration setting which of the aforementioned ambiguities is in use. Then the export can be in something reasonable like ISO 8601.

That said, dates that don't "parse" could be exported as "undefined" or equivalent.

**Reply by peter88213** [reply]
> Then NovelWriter could export that data, either in a format Timeline can use or a separate tool could transform something like a JSON export into a file Timeline understands.

Yes, this is possible for export. However, the real benefit would be synchronization with the timeline software so that it can be used for data entry. In my experience, it is quite convenient to edit the date/time data by dragging the mouse.
Synchronization works not via an intermediate format, but by directly overwriting the actual files.

**Reply by peter88213** [reply]
> I don't know enough Python to answer this but one hopes there are libraries designed to tackle the "all dates look different, even if you only use English" problem. 

I think, parsing the date/time/duration would be a task of the sync software, so everything is possible, as long as it writes the data back to novelWriter in the same format.

**Comment by instancezero** [comment]
I've just pushed https://github.com/instancezero/NovelWriterExtract as a first cut at extracting this metadata from a novelWriter project, essentially as a proof of concept.
It's a PHP command line thing. I looked for libraries in Python and Rust that were as capable as the ones available in PHP and found them wanting. Output to ODS or XSLX adds some formatting.

**Reply by vkbo** [reply]
If you were looking for libraries to write output files, [pandas](https://pandas.pydata.org/) is the Python package with best support for it as far as I know. It's used extensively in code for market analysis (what I do for work).

**Reply by vkbo** [reply]
I any case, thanks for the PoC. I'm looking into adding this for 2.5, which I have started working on, but as a starter I will probably only add support for CSV files. I don't want to pull in any of the big Python libraries, so ODS and XLSX support would have to be manually written. Thankfully, I know the Open Document standard fairly well.

**Reply by peter88213** [reply]
Here is a small Python script that writes csv files:

https://github.com/peter88213/nw_metadata

Have fun.

**Reply by vkbo** [reply]
CSV is trivial. Python has built-in support for it too. The issue is ODS and XLSX.

**Reply by vkbo** [reply]
Ah, sorry, didn't actually look at the code. This is for extracting meta data from novelWriter. Disregard my last reply 😆

**Reply by instancezero** [reply]
I took a look at pandas. While it will write the ODS files it doesn't add much over exporting CSV and importing it to one's spreadsheet of choice. As it's just an output for a generic data grid, there isn't much capability in formatting. My PoC script evaluates column widths, wraps long text cells, right-justifies columns that represent quantities, top-aligns the text, etc. etc. This makes the sheet immediately useful. All I'm missing is a little cell padding!
![Clipboard 1](https://github.com/vkbo/novelWriter/assets/2599327/8be72875-359e-4bee-9ce3-694c99d0ee76)

**Reply by peter88213** [reply]
Why not just edit the standard cell style in Calc? All right, then of course you can simply import csv. 

However, I have also written my own class library for reading and writing ODF formats with the right styles applied in order to be independent of third-party packages.

> This is for extracting meta data from novelWriter.

Well, this is a more generic approach. Could be improved, e.g. by sorting the headings, and using lookup tables like @instancezero does. Such lookup tables could be imported from project-specific INI files. Wouldn't that be a nice playground for the Python-savvy novelWriter user?

**Reply by vkbo** [reply]
I don't think it would take too much to expand my ODT writer to support ODS output. So it should be possible. I can split out the shared bits for the file format and put it into a new parent class, and implement the content writers as separate subclasses. I can look into that if I do implement a scene table view and add export for it.

I'm a little reluctant to open the can of worms that is Microsoft's document format as even MS Office can read Open Document files. Microsoft's maximalist approach to everything also applies to their document standard. It's an order of magnitude larger than the ODF one.

**Reply by peter88213** [reply]
One more thought: An advantage of ODS over CSV is the option of inserting hyperlinks. This even works between different ODF documents. 
So in my project, the generated ODS "plot grid" has hyperlinks that lead directly to the corresponding places in the generated ODT manuscript.

**Reply by instancezero** [reply]
There's little to no point in supporting a proprietary format. Excel can read ODS files. That's sufficient. The only reason why my extraction tool supports xlsx is because the library does and the code to exploit that is trivial. I haven't even bothered to test it because I don't have Excel.

Minimum formatting support for ODS is all that's required. As far as I can tell, export only needs to support cell alignment (horizontal and vertical), text wrap, and font bolding. Anything in addition to that is a bonus.

**Reply by vkbo** [reply]
It's actually not proprietary. It's a part of Office Open XML, which eventually was standardised. It's just a really horrible standard that only Microsoft could come up with. It should probably never have been accepted as a standard, and there was a big fight over that, but that's history as they say.

**Comment by vkbo** [comment]
Ok, so I have an idea how I want to implement the indexing of the story elements and handle them when processing the raw text.

Incidentally, I'm also wondering if the keyword should be `%Scene` instead of `%Story`. It is less generic, but it may make more sense with my second point below.

What I am a little concerned about is how to display dynamic categories in the Outline view. The table structure is not well suited for this as the number of columns can become rather large since there is no limit on the number of subcategories here. (Exporting them all is not a problem, and I'll just add switches for that in an export dialog to extend the current Outline export feature.)

Instead, I'm wondering if it is time to revisit the Outline view. I would like to create a new view that is scene-focused and displays information about each scene as a unit, and ignores the rest of the information. It doesn't need to necessarily replace the current Outline. I'm more thinking of adding a panel with a canvas and provide a collection of generators that can build various views and maps of the story in different formats. It could provide both timelines, scene cards focused on specific meta data, and various graphs even. If they are generated on a button click, it doesn't even matter if they are computationally costly to make.

I'm a little uncertain how to render them. I'm inclined to either use drawing capabilities of Qt to just generate SVG vector files and display them. There is [decent support for this in Qt](https://doc.qt.io/qt-5/qtsvg-svggenerator-example.html) already. I've considered webview as well, but I would like to avoid that I think. I would probably build it as a core class that can be extended to different kinds of views so that it is easier to contribute new views.

**Reply by vkbo** [reply]
Actually, about the `%Scene` vs `%Story`. In your proposal, `%Story` makes more sense. Maybe an equivalent on `%Scene` level would be useful though. Especially for the very technical writer who may like to specify `%Scene.Goal`, `%Scene.Conflict` etc.

**Reply by instancezero** [reply]
I'm laughing because I just pushed a change to make the "trigger keyword" easy to change. It wouldn't take much to recognize multiple triggers. When I'm in writer mode, I'm working with a scene, so %scene is the first thing that comes to mind but technically the data is about the overall story structure, which is why I originally used story.

I have been using the technical terms. I expect that once I've got this thing ready to send for a developmental edit my editor of choice will be looking at the story at that level, so I'm working through the same process myself.

At this point, the extractor recognizes and generates readable headings for these: `name, words, synopsis, value, polarity, purpose, incite, goal, complication, turning, crisis, climax, resolution, about, impact, pov, time, duration, others, pace, weather, appearance, touch, aural, smell, clothing, prose, emotions, comments`; as well as the `@char` and `@location` constructs (with more of the @ references to come; in this case `%story.pov` is intended to be used in the literary sense, e.g. "omniscient intercranial"). I plan to add a configuration file that allows the addition and override of the respective headings, providing user extensibility and headings in a different language.

As for presentation, in my branch, I just collected all the %story constructs and presented them in a group with highlighting similar to that of %synopsis, I also added a "structure" button at the bottom of the view panel to hide them. I'm not sure much else is required. I don't find that the technical information contributes a lot to the preview. The value of this information is really once it gets pulled into a table that spans all the scenes. In an ideal world that's done inside novelWriter and changes in the table get reflected in the corresponding document. I've considered making the exporter bidirectional but I think some sort of diff makes more sense... suggest changes in the structure by editing the sheet, then generate a "work list" that details the changes required. I know FreeCAD (https://github.com/FreeCAD/FreeCAD) has a Qt-based spreadsheet implementation that's likely overkill for this. Maybe there's a library somewhere that works? It would be extraordinary to be able to do this work without leaving novelWriter.

**Reply by vkbo** [reply]
I wasn't talking about how it's handled in the editor and viewer. That's trivial. I am concerned about the table view in the Outline and how to display that many columns. I think I would rather extend the lower panel to add a list of structure comments that is only visible when you select a scene.

Qt has a built-in table view. It is what the Outline used in the original implementation years ago. It looks awful. I absolutely do not want to pull in an external tool for this though. I try to keep novelWriter dependant on Qt only. (Aside from spell checking, which is anyway optional and novelWriter runs fine without it.)

A bi-directional feature would also violate a core principle of novelWriter: That no text will ever be modified by the code. If that is what we're looking for here, then this should be purely meta data and not part of the text documents.

In any case, the visualisation framework is a different feature, and implementing this does not depend on that. However, the information collected from `%Story` meta data should be available somewhere without having to export to a spreadsheet.

**Reply by vkbo** [reply]
I can, however, give a scene table view a try again as an alternate display mode in the Outline. It does have the flexibility of vertical expansion, which would be useful here. It could serve as a preview for an expanded meta data export feature.

The capabilities of the table view are described [here](https://doc.qt.io/qt-5/qtableview.html#details).

**Reply by instancezero** [reply]
That's a good core principle, and if the table view is part of novelWriter, then there's no friction in making changes in the document and then having it reflected in a refreshed table view. Then the only reason for an export is to send the data to an editor/agent in a form that they can access without learning a new tool. I would much prefer being able to stick inside novelWriter through the structural analysis process.

In an ideal world, the whole project is stored in a repo somewhere and my editor pushes changes/suggestions to it, then I merge/accept/deny/cry/rage as appropriate. Of the handful of people I'm considering for the editor role, I think only one of them could do that successfully. It's a cutting-edge edge case, or more likely an utter fantasy. The best case is I ship the project as a zip, then get it shipped back and do the diff myself.

**Reply by vkbo** [reply]
That's an interesting approach. I guess that's why some users have asked for a cloud storage feature of novelWriter, which I will absolutely never ever do! The project structure is designed to be file sync friendly and I think that's enough. I have considered built-in git integration, but the only people who'd use that are developers, and they already know how to do that most likely.

The scope of novelWriter is defined as a tool to arrive at a manuscript, and whatever happens in the editing phase is more or less out of scope. I have thought about a totally different approach for a potential new writing app, where everything is stored in a single document and all the tools just wrap around that. In such a setup, it would be possible to import annotations (comments) from an editor.

Maybe I'll find the time for that when I retire 😆

**Reply by instancezero** [reply]
I'm as averse to cloud systems as you are, but I think the primary use case is a different one, based on my conversations with other authors. Most of them want to be able to work from multiple devices without going to a so-called "cloud" solution where their work is stored on someone else's server. I know that's making a fine distinction between the two, but people seem to believe that files stored on a storage system are more protected than data buried in SomeSexyWriterApp.com.

**Reply by vkbo** [reply]
I work with cloud in my day job, and it makes sense in a work context. I want absolutely nothing to do with it for anything of my own data. I definitely don't want to be responsible for other people's data. So that's just a big *no*.

**Reply by instancezero** [reply]
I get it. I've moved my cloud storage from Dropbox to sync.com, but still have clients who share stuff over Dropbox/Google/etc. so I can't remove them from my systems.

Still, I hate to see authors who can barely afford it pay monthly subscription fees for some web-based writing app. I could see something like WEBDAV support. Most hosting companies can support that (hosting is my day job so I can say that with some confidence). At least in my case if they have a website, they have a place to have their work someplace that can be accessed from multiple devices.

Personally, that's all out of scope for me. If I'm mobile, I have my laptop with me!

**Reply by hifron** [reply]
So a tool to sync files to preferred solution even on client hosting is missing. But Disk Google is supported on many platforms and even OneDrive is also and set up where project is located in novelwriter is possible 🤫 and it is operating system thing and when application has support for it and enable such location in save dialog(or bookmark it) then even better... But knowing that needed such knowledge of operating system is advanced think when not necessary required for clients 🤤

**Reply by vkbo** [reply]
I would think most people who use fil sync tools know that there are folders on their PC that are synced and folders that are not? novelWriter is designed to be file sync friendly, in that it checks for file inconsistencies and project open in multiple locations. Aside from that, it doesn't care where the files are saved. Just like any other non-cloud app.

What I don't understand from your post, @hifron, is what tool you feel is missing.
